worker/src/RTC/NackGenerator.cpp

```diff
@@ -117,7 +118,7 @@ namespace RTC
 
 			// Do not let a packet pass if it's newer than last seen seq and came via
 			// RTX.
-			return false;
```

Comment:
I don't think this will ever happen, receiving a RTX with a seq num higher that the highest packet we've seen.

Reply:
I still don't know the reason. But program can easily run here by setting outgoing traffic 5% random loss on a webapp host 
(I use Network Emulator for Windows Toolkit to set random loss)

Please see ssrc:3650588773

 mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2507709574, seq:16626, recovered:false] +2ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2507709574, seq:16627, recovered:false] +6ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:3650588773, seq:30671, recovered:false] +22ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2393181301, seq:16498, recovered:false] +1ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2507709574, seq:16628, recovered:false] +0ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:3027678492, seq:30763, recovered:false] +4ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2507709574, seq:16629, recovered:false] +1ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:3027678492, seq:30764, recovered:false] +5ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2507709574, seq:16630, recovered:false] +3ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2507709574, seq:16631, recovered:false] +13ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:3027678492, seq:30765, recovered:false] +2ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2393181301, seq:16499, recovered:false] +4ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2507709574, seq:16632, recovered:false] +0ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:3650588773, seq:30672, recovered:false] +1ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:3027678492, seq:30766, recovered:false] +0ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2507709574, seq:16633, recovered:false] +5ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:3027678492, seq:30767, recovered:false] +7ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2507709574, seq:16634, recovered:false] +0ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:3027678492, seq:30768, recovered:false] +6ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2393181301, seq:16500, recovered:false] +13ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2507709574, seq:16635, recovered:false] +0ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2507709574, seq:16636, recovered:false] +12ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2507709574, seq:16637, recovered:false] +6ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2507709574, seq:16638, recovered:false] +6ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:3650588773, seq:30673, recovered:false] +9ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:3027678492, seq:30769, recovered:false] +6ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:3027678492, seq:30770, recovered:false] +6ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:3027678492, seq:30771, recovered:false] +6ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:3027678492, seq:30772, recovered:false] +6ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2393181301, seq:16501, recovered:false] +8ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2507709574, seq:16639, recovered:false] +0ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:3027678492, seq:30773, recovered:false] +3ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2507709574, seq:16640, recovered:false] +3ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:2507709574, seq:16641, recovered:false] +6ms
  mediasoup:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | NACKed packet received [ssrc:3650588773, seq:30674, recovered:true] +2ms
  mediasoup:WARN:Channel [pid:3283] RTC::NackGenerator::ReceivePacket() | newer than last seen seq [ssrc:3650588773, seq:30674, last:30673] +246ms

Reply:
I still don't know the reason. But the program can easily run here by setting 5% random loss on webapp host

Reply:
Please @bianxg provide much more info. The `NackGenerator` code has been reviewed tons of times. I cannot believe that the changes you propose make sense at all. Please upload comparative videos or stats or whatever.

Reply:
[1.txt](https://github.com/versatica/mediasoup/files/5724511/1.txt)
It's very easy to reproduce this(no more than several seconds):
1 I use edumeet to test
2 set webapp outbound loss rate 5% using https://jagt.github.io/clumsy/
![1](https://user-images.githubusercontent.com/3364079/102782344-2b173100-43d4-11eb-9b78-ebb25d3e82e1.png)
3 add a MS_ASSERT at here
             ```
               if (isRecovered)
		{
			this->recoveredList.insert(seq);

			// Remove old ones so we don't accumulate recovered packets.
			auto it = this->recoveredList.lower_bound(seq - MaxPacketAge);

			if (it != this->recoveredList.begin())
				this->recoveredList.erase(this->recoveredList.begin(), it);

			// Do not let a packet pass if it's newer than last seen seq and came via
			// RTX.
			MS_ASSERT(false, "newer than last seen seq");
			return true;
		}
```

Reply:
How can the browser send via RTX a packet whose original Seq number is higher than anyone previously received by mediasoup.

BTW @jmillan I've seen this in my tests, this is, the `return true` in comment above. So this is happening. Wondering, is the browser crazy or are we doing a wrong check before it?

Reply:
I can't neither say. We need to capture the traffic and check.

Reply:
> I can't neither say. We need to capture the traffic and check.

I'll give it a try.

NOTE this is not relevant to the issue.